tests/test_FastApiPacker.py

```diff
@@ -0,0 +1,258 @@
+    os.makedirs(framework_dir)
+    (framework_dir / "run.sh").write_text("#!/bin/bash")
+
+    metadata_file = temp_model_directory / "metadata.json"
+    metadata_file.write_text(json.dumps({"Identifier": "test_model_id"}))
+
+    # Correctly formatted install.yml file with a valid python version
+    install_file = temp_model_directory / "install.yml"
+    install_file.write_text("""
+python: "3.8"
+pip:
+  - requests
+commands:
+  - pip install -r requirements.txt
+""")
+
+    # Print to check the contents of the YAML file
+    with open(install_file, "r") as f:
+        print(f"YAML file content: {f.read()}")  # To check if it's being read correctly
+
+    # Initialize the FastApiAppPacker and check the install_writer
+    try:
+        packer = FastApiAppPacker(str(temp_model_directory), str(bundles_repo_path))
+        print(f"Install writer: {packer.install_writer}")  # Check if the install writer is created
+
+        # Call the method to be tested after successful initialization
+        packer._create_app_files()  # Ensure this method is called after successful initialization
+        packer._edit_post_commands_app()  # Add this if it's meant to be called
+
+    except ValueError as e:
+        print(f"Error: {str(e)}")  # Catch and print any ValueError raised during initialization
+        # Handle exception as needed (e.g., re-raise if necessary or assert the expected error)
+
+    # Check if 'open' was called with the correct file and mode (read)
+    mock_open_file.assert_called_with(str(install_file), "r")  # Convert PosixPath to string
+
+
+
+@patch("builtins.open", new_callable=mock_open)
+def test_write_api_schema(mock_open_file, temp_model_directory, bundles_repo_path):
+    # Write the metadata file content
+    metadata_file = temp_model_directory / "metadata.json"
+    metadata_content = json.dumps({"Identifier": "test_model_id"})
+    metadata_file.write_text(metadata_content)
+
+    # Write the install.yml file content
+    install_file = temp_model_directory / "install.yml"
+    install_content = yaml.dump({"python": "3.8", "pip": ["requests"]})
+    install_file.write_text(install_content)
+
+    # Configure the mock to return the content for both files
+    def mock_open_side_effect(file, mode="r", *args, **kwargs):
+        if file == str(metadata_file):
+            return mock_open(read_data=metadata_content).return_value
+        elif file == str(install_file):
+            return mock_open(read_data=install_content).return_value
+        else:
+            raise FileNotFoundError(f"Mock not configured for file: {file}")
+
+    mock_open_file.side_effect = mock_open_side_effect
+
+    # Create the FastApiAppPacker instance
+    packer = FastApiAppPacker(str(temp_model_directory), str(bundles_repo_path))
+    
+    # Add assertions to verify functionality (e.g., check the generated bundle directory)
+    assert packer.model_id == "test_model_id"
+
+
+# Run tests
+if __name__ == "__main__":
+    pytest.main()
```

Comment:
We don't really need this to run the tests as a script. Pytest's test discovery will take care of finding these tests within the repo.